library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -357,14 +376,15 @@ class Client() {
             }
 
         if (v3Client != null) {
-            if (v3Client.textToSign() == null) {
-                v3Client.registerIdentity(null)
-            } else if (account != null) {
-                v3Client.textToSign()?.let {
-                    v3Client.registerIdentity(account.sign(it)?.rawData)
+            v3Client.signatureRequest()?.let { signatureRequest ->
+                if (account != null) {
+                    account.sign(signatureRequest.signatureText())?.let {
+                        signatureRequest.addEcdsaSignature(it.rawData)
+                    }
+                    v3Client.registerIdentity(signatureRequest)
+                } else {
+                    throw XMTPException("No signer passed but signer was required.")
                 }
```

Comment:
I'm not totally sure on how this is supposed to work now. Would love some guidance. 🙏

Reply:
I'm getting this error from libxmtp `org.xmtp.android.library.XMTPException: Error creating client Unknown signer`

Reply:
I can take a quick look into that. That error is coming from the original `createClient`? Or it's happening down here?

Reply:
I created a [test case](https://github.com/xmtp/libxmtp/pull/767) that reproduces an error with legacy keys. Not sure if that error is the same one you're facing.